src/Compilers/CSharp/Portable/Declarations/DeclarationTreeBuilder.cs

```diff
@@ -619,7 +682,56 @@ public override SingleNamespaceOrTypeDeclaration VisitEnumDeclaration(EnumDeclar
                 nameLocation: new SourceLocation(node.Identifier),
                 memberNames: memberNames,
                 children: ImmutableArray<SingleTypeDeclaration>.Empty,
-                diagnostics: diagnostics.ToReadOnlyAndFree());
+                diagnostics: diagnostics.ToReadOnlyAndFree(),
+                _nonGlobalAliasedQuickAttributes | quickAttributes);
+        }
+
+        private QuickAttributes GetQuickAttributes(SyntaxList<AttributeListSyntax> attributeLists)
+        {
+            var result = QuickAttributes.None;
+            foreach (var attributeList in attributeLists)
+            {
+                foreach (var attribute in attributeList.Attributes)
+                    result |= GetQuickAttributes(attribute.Name.GetUnqualifiedName().Identifier.ValueText);
+            }
+
+            return result;
+
+        }
+
+        private static QuickAttributes GetQuickAttributes(string name)
+        {
+            var result = QuickAttributes.None;
+            if (Matches(name, AttributeDescription.TypeIdentifierAttribute))
+                result |= QuickAttributes.TypeIdentifier;
+
+            if (Matches(name, AttributeDescription.TypeForwardedToAttribute))
+                result |= QuickAttributes.TypeForwardedTo;
+
+            if (Matches(name, AttributeDescription.AssemblyKeyNameAttribute))
+                result |= QuickAttributes.AssemblyKeyName;
+
+            if (Matches(name, AttributeDescription.AssemblyKeyFileAttribute))
+                result |= QuickAttributes.AssemblyKeyFile;
+
+            if (Matches(name, AttributeDescription.AssemblySignatureKeyAttribute))
+                result |= QuickAttributes.AssemblySignatureKey;
+
+            return result;
+
+            static bool Matches(string name, AttributeDescription attributeDescription)
+            {
+                if (name == attributeDescription.Name)
+                    return true;
+
+                if ((name.Length + nameof(System.Attribute).Length) == attributeDescription.Name.Length &&
```

Comment:
It looks like we are using the same check for attribute classes in attribute lists as well as attribute classes in using aliases. If I am reading correctly, it seems like this function would return true for something like `using X = AssemblyKeyFile`, when it's actually necessary for the right side to be `AssemblyKeyFileAttribute` for it to refer to the attribute. Right?

Reply:
that is correct.  but this is simply a weak heuristic.  if someone happens to go `using X = AssemblyKeyFile` and that causes us to go back to source that's fine, it will be a little extra work.  However, as this will basically never happen in practice, there's no practical downside to this approach.

Reply:
though it would be pretty simple to add a boolean if we need to do this work or not.  but it doesn't really seem that important to me.

Reply:
ok.  i did this :)